harvester/DBInterface.py

```diff
@@ -25,11 +24,11 @@ def __init__(self, params):
 
 		elif self.dbtype == "postgres":
 			self.dblayer = __import__('psycopg2')
-			con = self.getConnection()
 
 		else:
 			raise ValueError('Database type must be sqlite or postgres in config file')
 
+		con = self.getConnection()
 		with con:
```

Comment:
I'm assuming we can do this now because the imports are handled elsewhere and this code is no longer DB interface-specific? It seems like it's partially related to your other cleanups moving out some of the DB logic, correct me if I'm wrong...